src/main/java/com/google/idea/perf/methodtracer/TracerMethodTransformer.kt

```diff
@@ -152,13 +153,18 @@ class TracerMethodTransformer : ClassFileTransformer {
                     }
 
                     private fun loadArgSet() {
-                        if (parameters == 0) {
+                        var arraySize = 0
+                        for (parameterIndex in parameterTypes.indices) {
+                            if (parameters and (1 shl parameterIndex) != 0) {
+                                arraySize++
+                            }
+                        }
+
+                        if (arraySize == 0) {
                             mv.visitInsn(ACONST_NULL)
                             return
                         }
 
-                        val arraySize = Integer.bitCount(parameters)
```

Comment:
Hm, I think you lost a change here?

Reply:
This was intentional to handle a corner case where a parameter index is out of bounds.

Reply:
Ahh, makes sense!